internal/xds/translator/translator.go

```diff
@@ -512,3 +534,82 @@ func addXdsCluster(tCtx *types.ResourceVersionTable, args *xdsClusterArgs) error
 	}
 	return nil
 }
+
+const (
+	DefaultEndpointType EndpointType = iota
+	Static
+	EDS
+)
+
+func buildXdsUpstreamTLSCertSecret(tlsConfig ir.TLSBundle) *tlsv3.Secret {
```

Comment:
can we rename and reuse https://github.com/envoyproxy/gateway/blob/d0c12ebed88bf4f6244230e96ebc9c1670517b1a/internal/xds/translator/listener.go#L454 ?

Reply:
yes, will update

Reply:
can we use `buildXdsTLSSecret` as the function name ?

Reply:
sure sg